cli/tests/rsa_signature.go

```diff
@@ -0,0 +1,409 @@
+						// Seed the random number generator
+						// #nosec G404
+						seed := time.Now().UnixNano() // Using current time as a seed for demonstration
+						// #nosec G404
+						localRand := rand.New(rand.NewSource(seed))
+
+						// Generate a random number, for example between 0 and 9
+						// #nosec G404
+						dynamicValue := localRand.Intn(10) // This will give you a random integer between 0 and 9
+
+						// Using fmt.Sprintf to insert the dynamic value into the string
+						signatureString := fmt.Sprintf("sVV0fsvqQlqliKv/ussGIatxpe8LDWhc8uoaM5WpjbiYvvxUr1eCpz0ja7UT%dPGNDdmoGi6xbMC1g/ozhAt4uCdpy0Xdfqbv", dynamicValue)
+
+						// Converting the formatted string to a byte slice
+						signature := []byte(signatureString)
+						mm.Signature = signature
+						incomingMessage.Data, err = mm.Encode()
+						incomingMessage.Body, err = incomingMessage.Encode()
+						if err != nil {
+							logger.Fatal("failed to encode message", zap.Error(err))
+						}
+
+						if sendMessageCount < 4 {
+							err = p2pNetwork.Broadcast(ssvMessage)
+							if err != nil {
+								logger.Fatal("failed to broadcast message", zap.Error(err))
+							}
+							sendMessageCount++
+						}
+					}
+				}
+			}
+		}()
+		err = p2pNetwork.SubscribeAll(logger)
+		if err != nil {
+			logger.Fatal("can't subscribe all (rsa)", zap.Error(err))
+		}
+		p2pNetwork.UpdateSubnets(logger)
+	},
+}
+
+func setupOperatorStorage(logger *zap.Logger, db basedb.Database) (operatorstorage.Storage, *registrystorage.OperatorData) {
+	nodeStorage, err := operatorstorage.NewNodeStorage(logger, db)
+	if err != nil {
+		logger.Fatal("failed to create node storage", zap.Error(err))
+	}
+
+	cfg.P2pNetworkConfig.OperatorPrivateKey, err = decodePrivateKey(cfg.OperatorPrivateKey)
+	if err != nil {
+		logger.Fatal("could not decode operator private key", zap.Error(err))
+	}
+
+	operatorPubKey, err := nodeStorage.SetupPrivateKey(cfg.OperatorPrivateKey)
+	if err != nil {
+		logger.Fatal("could not setup operator private key", zap.Error(err))
+	}
+
+	_, found, err := nodeStorage.GetPrivateKey()
+	if err != nil || !found {
+		logger.Fatal("failed to get operator private key", zap.Error(err))
+	}
+	var operatorData *registrystorage.OperatorData
+	operatorData, found, err = nodeStorage.GetOperatorDataByPubKey(nil, operatorPubKey)
+	if err != nil {
+		logger.Fatal("could not get operator data by public key", zap.Error(err))
+	}
+	if !found {
+		operatorData = &registrystorage.OperatorData{
+			PublicKey: operatorPubKey,
+			ID:        4,
+		}
+	}
+
+	return nodeStorage, operatorData
+}
+
+func decodePrivateKey(key string) (*rsa.PrivateKey, error) {
+	operatorKeyByte, err := base64.StdEncoding.DecodeString(key)
+	if err != nil {
+		return nil, err
+	}
+
+	sk, err := rsaencryption.ConvertPemToPrivateKey(string(operatorKeyByte))
+	if err != nil {
+		return nil, err
+	}
+
+	return sk, err
+}
+
+func setupDB(logger *zap.Logger, eth2Network beaconprotocol.Network) (*kv.BadgerDB, error) {
```

Comment:
do we even need the db?

Reply:
yes we do need for nodeStorage that been using in p2p